tfhe/src/core_crypto/algorithms/test/lwe_programmable_bootstrapping.rs

```diff
@@ -165,6 +165,214 @@ where
 
 create_parametrized_test!(lwe_encrypt_pbs_decrypt_custom_mod);
 
+// Here we will define a helper function to generate a many lut accumulator for a PBS
+fn generate_accumulator_many_lut<Scalar: UnsignedTorus + CastFrom<usize>>(
+    polynomial_size: PolynomialSize,
+    glwe_size: GlweSize,
+    message_modulus: usize,
+    ciphertext_modulus: CiphertextModulus<Scalar>,
+    delta: Scalar,
+    funcs: &[&dyn Fn(Scalar) -> Scalar],
+) -> (GlweCiphertextOwned<Scalar>, usize) {
```

Comment:
We do not require the power of 2 in that case it's just that we use power of 2 to organize the data which IMO is much less of a headache

Reply:
But it seems to me that it limits the maximum possible degree of the inputs of the many LUT

Reply:
I think the first implementation will be based on the power of 2 approach, and if we see we are missing some optimizations we'll likely update to be able to be more flexible wrt to the degree.

The only time we would have a limit is if we have a small degree and lots of functions then there may be a world where some functions will not be evaluated because of wasted space if we use the power of 2, I think it's an acceptable tradeoff at first.

Plus here for core we don't really care, it's a first implementation to check the principles, the question will be of more interest later when it goes up to shortint

Reply:
well at least from a design perspective it means the LUT generation does not need the input ciphertext degree in the power of 2 approach as it's a pessimistic case, in the degree approach the lut needs to be taylor made for the ciphertext

Reply:
As described in my second comment, I believe we can take the biggest possible `input_max_degree` (which depend on `fn_count` and `full_msg_mod`) to build the LUT and we can use it for all compatible ciphertexts